accounts/static/accounts/js/onboarding_movies.js

```diff
@@ -0,0 +1,522 @@
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error("Network response was not ok");
+      }
+
+      window.location.href = "/accounts/onboarding/triggers/";
+    })
+    .catch((error) => {
+      console.error("Error submitting movie preferences:", error);
+    });
+}
+
+function redirectToOnboardingGenreForm() {
+  window.location.href = "/accounts/onboarding/genres/";
+}
+
+function openBackModal() {
+  document.getElementById("confirm_back_modal").classList.remove("hidden");
+  document.getElementById("exit_overlay").classList.remove("hidden");
+}
+
+function closeBackModal() {
+  document.getElementById("confirm_back_modal").classList.add("hidden");
+  document.getElementById("exit_overlay").classList.add("hidden");
+}
+
+/************************************
+ *
+ *  Page initialization
+ *
+ ************************************/
+/**
+ * This function is called when the page loads, it populates the user data
+ * sets so that the page may be styled accordingly based on current user preferences.
+ */
+async function loadUserMovies() {
+  const csrfToken = await getCSRFToken();
+  fetch("/accounts/preferences/movies/", {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json",
+      "X-CSRFToken": csrfToken,
+    },
+  })
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error("Network response was not ok");
+      }
+
+      return response.json();
+    })
+    .then((data) => {
+      movies_liked = extractIds(data.movies_liked);
+      movies_disliked = extractIds(data.movies_disliked);
+      movies_watched = extractIds(data.movies_watched);
+      watchlist = extractIds(data.watchlist);
+      movies_rewatch = extractIds(data.movies_rewatch);
+      movies_blocked = extractIds(data.movies_excluded);
+    });
+}
+
+function extractIds(movieList) {
+  return new Set(movieList.map((movie) => movie.pk));
+}
+
+// Random movie fetch when scrolling past halfway point
+document.addEventListener("DOMContentLoaded", function () {
+  movieContainer.addEventListener("scroll", () => {
+    searchString = searchbar.value.trim();
+    threshold = (movieContainer.scrollHeight * 3) / 4;
+    if (!searchString && movieContainer.scrollTop >= threshold) {
+      console.log("Fetching more movies...");
+      fetchMovies();
+    }
+  });
+});
+
+// Hides the tooltip when the user clicks outside of it
+document.addEventListener("click", function (event) {
+  if (currentTooltiptext && !currentTooltiptext.contains(event.target)) {
+    currentTooltiptext.style.display = "none";
+  }
+});
+
+/************************************
+ *
+ * User preference button functions
+ *
+ ************************************/
+
+/**
+ * This function attempts to update the liked status of a movie.
+ * @param {String} id the id of the movie to be updated
+ */
+function addLiked(id) {
+  liked_button = document.getElementById(id + "_upvote");
+  intID = parseInt(id);
+  if (movies_liked.has(intID)) {
```

Comment:
odd new line for the return
if it's returning, then don't make the other bit a 'else if' especially if they're not going to

if {}

if {}
else {}

rest of code